### PR TITLE
[ci] Reset Latest Release

### PR DESCRIPTION
**Pull request by ppenna**
This pull request introduces improvements to the release workflow and refactors parts of the `nanvix-registry` library for clarity and correctness. The main changes include adding a new workflow step to reset the "latest" release and tag before updating, disabling updates to existing releases, and cleaning up variable names and logic in the registry code. Some redundant or problematic tests have also been removed.

**Release workflow improvements:**

* Added a new "Reset Latest Release" step in `.github/workflows/self-hosted-ci.yml` to delete the existing "latest" release and tag before creating a new one, ensuring the release process is clean and avoids conflicts.
* Set `allowUpdates: false` in the release action configuration to prevent updates to existing releases, enforcing that each release is newly created.

**Registry code refactoring and cleanup:**

* Renamed the `release_registry` variable to `registry` throughout `src/libs/nanvix-registry/src/lib.rs` for consistency and clarity. [[1]](diffhunk://#diff-39070155be1ab8a72d4934fc13965effe2918c993bccf5713ef89c9916342834L417-R417) [[2]](diffhunk://#diff-39070155be1ab8a72d4934fc13965effe2918c993bccf5713ef89c9916342834L434-R434) [[3]](diffhunk://#diff-39070155be1ab8a72d4934fc13965effe2918c993bccf5713ef89c9916342834L472-R473)
* Simplified the logic in `extract_commit_id` to remove unreachable code and improve readability.

**Test cleanup:**

* Removed tests for edge cases that are no longer relevant or supported, such as full 40-character commit IDs and empty commit IDs in release URLs. [[1]](diffhunk://#diff-39070155be1ab8a72d4934fc13965effe2918c993bccf5713ef89c9916342834L1045-L1050) [[2]](diffhunk://#diff-39070155be1ab8a72d4934fc13965effe2918c993bccf5713ef89c9916342834L1063-L1068)

### CODE REVIEW

File: .github/workflows/self-hosted-ci.yml [hunk 1]
```diff
@@ -250,6 +250,60 @@ jobs:
...
+              import json
+              import sys
+
+              with open(sys.argv[1], 'r', encoding='utf-8') as handler:
+                data = json.load(handler)
+
+              release_id = data.get('id')
+              print(release_id if release_id is not None else '')
+              PY
+              )
```
[the remark this answers is not in the document]
Comment (ppenna) [reply .github/workflows/self-hosted-ci.yml:287]: Indentation is correct.
